### ISSUE TITLE
Incompatible with 1.21.1

### ISSUE THREAD

**Issue by ColonalLive** [tr7zw/EntityCulling#164]
Version 1.6.6 release is incompatible with 1.21.1.
Just want to express interest for this mod to be updated when you get the chance :)

![image](https://github.com/user-attachments/assets/155d968c-358c-46ef-bb28-17c97b5a269f)

**Comment by mucookul** [comment]
There are no major changes, it's like 1.20.5 to 1.20.6, use a dependency override for now

**Comment by amongsus1614** [comment]
I even changed the files and in them the versions from 1.21 to 1.21.1 but I still have the same problem

**Comment by amongsus1614** [comment]
On qulit is same problem

**Comment by tr7zw** [comment]
I'm not at home currently. Also noteworthy that this only affects fabric(forge/neoforge treat 1.21.1 as 1.21) and that there is no reason or benifit to update clients to 1.21.1. So for now till I have time to roll out updates to all mods, just ignoring 1.21.1 is the solution. (or dependency override/changing the json, if you really want to use 1.21.1 for whatever reason)

**Comment by amongsus1614** [comment]
No override .json don t work

**Comment by amongsus1614** [comment]
he
![screen](https://github.com/user-attachments/assets/ac221d8f-c4be-4e7d-8319-f60f9c01ff4a)
re is the proof

**Comment by Razuuu** [comment]
> he ![screen](https://private-user-images.githubusercontent.com/178566573/358565748-ac221d8f-c4be-4e7d-8319-f60f9c01ff4a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NDM1MDcsIm5iZiI6MTcyMzk0MzIwNywicGF0aCI6Ii8xNzg1NjY1NzMvMzU4NTY1NzQ4LWFjMjIxZDhmLWM0YmUtNGU3ZC04MzE5LWY2MGY5YzAxZmY0YS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOFQwMTA2NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNDcwNmRmZjllMmU5NDc5OTQxZDYyZDNlZWU4OTE5MjQwYmZiMmM4ZWU4OTUzZGI2MGRmMDZhNjQ1YjllNjA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yqYKcmUc8M7fnN-YSJGMqSL3zz3wrZ9j4oQxipwKizA) re is the proof

If you edited the file in Ark, it won't save the changes properly. You need to copy the file to a folder, make the change to 1.21.1 and copy it back into the JAR file.
or use this instead:         "minecraft": ">=1.21",

**Comment by litetex** [comment]
Workaround until this is fixed:
1. Extract the jar file
2. Open ``fabric.mod.json`` inside the unzipped directory
3. Remove the line that contains ``"minecraft": "1.21"``
4. Rezip the contents (make sure that you don't have a nested directory in the zip!)
5. Rename the .zip to .jar

or use my file where I already did this:
[entityculling-fabric-1.6.6-mc1.21.jar.zip](https://github.com/user-attachments/files/16663682/entityculling-fabric-1.6.6-mc1.21.jar.zip) (GitHub only supports .zip files so you have to extract the .jar file!)

Recommended fix for the future from my side:
Instead of ``depending`` on a version simply ``recommend`` it.
Basically remove this line: https://github.com/tr7zw/EntityCulling/blob/5e3c5c3654ca5196cf9d485c024306b45a6339f5/src/main/resources/fabric.mod.json#L25
And add
```
	"recommends": {
		"minecraft": "${minecraft_version_range_fabric}"
	},
```
instead.
This doesn't cause a hard crash and instead shows a warning inside the log.

**Comment by Razuuu** [comment]
> Workaround until this is fixed:
> 
>     1. Extract the jar file
> 
>     2. Open `fabric.mod.json` inside the unzipped directory
> 
>     3. Remove the line that contains `"minecraft": "1.21"`
> 
>     4. Rezip the contents (make sure that you don't have a nested directory in the zip!)
> 
>     5. Rename the .zip to .jar
> 
> 
> or use my file where I already did this: [entityculling-fabric-1.6.6-mc1.21.jar.zip](https://github.com/user-attachments/files/16663682/entityculling-fabric-1.6.6-mc1.21.jar.zip) (GitHub only supports .zip files so you have to extract the .jar file!)
> 
> Recommended fix for the future from my side: Instead of `depending` on a version simply `recommend` it. Basically remove this line:
> 
> https://github.com/tr7zw/EntityCulling/blob/5e3c5c3654ca5196cf9d485c024306b45a6339f5/src/main/resources/fabric.mod.json#L25
> 
> And add
> 
> ```
> 	"recommends": {
> 		"minecraft": "${minecraft_version_range_fabric}"
> 	},
> ```
> 
> instead. This doesn't cause a hard crash and instead shows a warning inside the log.

https://github.com/Razuuu/EntityCulling/releases/download/1.6.7/entityculling-fabric-1.6.7-mc1.21.jar
My release fixes start with 1.21.1 - just download and paste it
